Fixed the tests with C++11 or newer, and extended the travis config

dutow opened the pull request:
Currently the build with clang, std=c++17 and libcxx fails, which is addressed by my other PR.

I also disabled two warnings in these builds, one is the c++11+ specific null-nullptr warning, the other is about marco names strarting with two underscores.

nickbruun commented:
Let's get this rebased and see if we get a pass. Having a test for C++17 is a great idea.

dutow commented:
I rebased it and also added the llvm lib directory to the LD_LIBRARY_PATH - without that the test can't find the libc++ runtime and fails to start up (as a fixed step, as the test only uses one clang version and it doesn't affect gcc)

nickbruun commented:
Awesome stuff. Thanks once again, @dutow!
